cloudformation/auxiliary_templates/custom_balancer/custom_queue_job_balancer.yaml

```diff
@@ -0,0 +1,436 @@
+
+  UnprocessedBalancingEventQueue:
+    Type: 'AWS::SQS::Queue'
+    Properties:
+      QueueName: !Join
+        - "-"
+        - - deadline-unprocessed-balancing-events
+          - !Ref QueueId
+    UpdateReplacePolicy: Delete
+    DeletionPolicy: Delete
+
+  UnprocessedBalancingEventQueuePolicy:
+    DependsOn: DeadlineWeightedJobBalancerSchedule
+    Type: 'AWS::SQS::QueuePolicy'
+    Properties:
+      PolicyDocument:
+        Statement:
+          - Action: 'sqs:SendMessage'
+            Condition:
+              ArnEquals:
+                'aws:SourceArn': !GetAtt
+                  - DeadlineWeightedJobBalancerSchedule
+                  - Arn
+            Effect: Allow
+            Principal:
+              Service: events.amazonaws.com
+            Resource: !GetAtt
+              - UnprocessedBalancingEventQueue
+              - Arn
+        Version: 2012-10-17
+      Queues:
+        - !Ref UnprocessedBalancingEventQueue
+
+  DeadlineWeightedJobBalancerExecutionRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - lambda.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      RoleName: !Join
+        - ''
+        - - !Ref 'AWS::StackName'
+          - DeadlineWeightedJobBalancerRole
+      Policies:
+        - PolicyName: DefaultExecutionRole
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action: logs:CreateLogGroup
+                Resource: !Sub 'arn:aws:logs:${AWS::Region}:${AWS::AccountId}:*'
+              - Effect: Allow
+                Action:
+                  - logs:CreateLogStream
+                  - logs:PutLogEvents
+                Resource: !Sub "arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/lambda/*:*:*"
+        - PolicyName: LambdaPermissions
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - deadline:GetFleet
+                  - deadline:UpdateJob
+                  - deadline:SearchJobs
+                  - deadline:ListSteps
+                  - deadline:ListTasks
+                  - deadline:ListQueueFleetAssociations
+                Resource:
+                  - !Sub "arn:aws:deadline:${AWS::Region}:${AWS::AccountId}:farm/${FarmId}"
+                  - !Sub "arn:aws:deadline:${AWS::Region}:${AWS::AccountId}:farm/${FarmId}/fleet/*"
+                  - !Sub "arn:aws:deadline:${AWS::Region}:${AWS::AccountId}:farm/${FarmId}/queue/${QueueId}/job/*"
+                  - !Sub "arn:aws:deadline:${AWS::Region}:${AWS::AccountId}:farm/${FarmId}/queue/${QueueId}"
+              - Effect: Allow
+                Action:
+                  - identitystore:ListGroupMembershipsForMember
+                Resource: '*'
```

Comment:
This looks like an unnecessary statement, I don't see the API being called.

Reply:
It looks that this is a Dependent action of `deadline:GetFleet`

https://docs.aws.amazon.com/service-authorization/latest/reference/list_awsdeadlinecloud.html

Reply:
I suspect it isn't necessary though, because the policy isn't doing any condition key authorization based on membership within the Lambda. If we can remove it, I think we should.